### PR TITLE
[fix #361] disable snapping on invisible features

### LINKED ISSUE #361: qfield snapping: invisible features should be ignored
If snapping on a layer with rulebased-render also invisible features are snapped.
In QGIS this is not the case and also very usefull for digitizing.

### Reproduction Steps:

1. QGIS: Enable snapping on layer with categorized render style (some features visible/some not)
2. QFIELD: Snap on invisible feature

### Expected behavior:

Only visible features are snappable

### Observed behavior:

All features incl. invisible ar snappable

**QField version: Lucendro (0.11.6)** 

## Additional information:

* Problem started happening recently, didn't happen in an older version of QField: [No]
* Problem can be reliably reproduced, doesn't happen randomly: [Yes]
* Problem happens with all files and projects, not only some files or projects: [Yes]

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-678-3465bfed0b33d3ebd7057c9eacad1334fc9bc7dd-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-678-3465bfed0b33d3ebd7057c9eacad1334fc9bc7dd-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-678-3465bfed0b33d3ebd7057c9eacad1334fc9bc7dd-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-678-3465bfed0b33d3ebd7057c9eacad1334fc9bc7dd-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-678-a6d393639a700083e35e04c968d20a20549bd270-armv7.apk)
 - [arm64_v8a](https://download.opengis.ch/qfield/ci-builds/qfield-dev-678-a6d393639a700083e35e04c968d20a20549bd270-arm64_v8a.apk)
 - [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-678-a6d393639a700083e35e04c968d20a20549bd270-x86.apk)
 - [x86_64](https://download.opengis.ch/qfield/ci-builds/qfield-dev-678-a6d393639a700083e35e04c968d20a20549bd270-x86_64.apk)

File: src/core/snappingutils.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@
 #include "qgsproject.h"
 
 SnappingUtils::SnappingUtils( QObject *parent )
-  : QgsSnappingUtils( parent )
+  : QgsSnappingUtils( parent, false /*enableSnappingForInvisibleFeature*/ )
```
Comment (signedav) [inline src/core/snappingutils.cpp]:
```suggestion
  : QgsSnappingUtils( parent, false )
```
Is this comment only to remember, that we could have a setting for that in future?
If so, then I would remove it, and maybe open an enhancement-request for it's implementation instead.
But up to you.

Comment (signedav) [reply src/core/snappingutils.cpp]: So no removing :slightly_smiling_face:

Comment (3nids) [reply src/core/snappingutils.cpp]: it's to know what this is used for without having to look in the API docs.
or why using flags instead of booleans.

Comment (signedav) [reply src/core/snappingutils.cpp]: Okay. It's fine.

**Pull request merged**